File: Estadistica.java
import java.util.List;

class Estadisticas {
    private List<Double> datos;

    public void setDatos(List<Double> datos) {
        this.datos = datos;
    }
    public List<Double> getDatos() {
        return datos;
    }
    public double promedio() {
        double suma = 0;
        for (double num : datos) {
            suma += num;
        }
        return suma / datos.size();
    }
    
    public double desviacion() {
        double prom = promedio();
        double suma = 0;
        for (double num : datos) {
            suma += Math.pow(num - prom, 2);
        }
        return Math.sqrt(suma / (datos.size() - 1));
    }
    
    public String toString() {
        return String.format("El promedio es %.2f\nLa desviación estándar es %.6f", promedio(), desviacion());
    }
    
    public static void main(String[] args) {
        List<Double> valores = List.of(1.9, 2.5, 3.7, 2.0, 1.0, 6.0, 3.0, 4.0, 5.0, 2.0);
        Estadisticas stats = new Estadisticas();
        stats.setDatos(valores);
        System.out.println(stats);
    }

File: Estadistica.py
import math
class Estadisticas:
    datos = []
    def setDatos(self, datos):
        self.datos = datos
    def getDatos(self):
        return self.datos
    def promedio(self):
        return sum(self.datos) / len(self.datos)
    def desviacion(self):
        prom = self.promedio()
        suma = sum((x - prom) ** 2 for x in self.datos)
        return math.sqrt(suma / (len(self.datos) - 1))
    
    def __str__(self):
        return f"El promedio es {self.promedio():.2f}\nLa desviación estándar es {self.desviacion():.6f}"

valores = [1.9, 2.5, 3.7, 2, 1, 6, 3, 4, 5, 2]
stats = Estadisticas()
stats.setDatos(valores)
print(stats)
